src/phase0/state_transition/epoch_processing.rs

```diff
@@ -888,14 +893,9 @@ pub fn get_inactivity_penalty_deltas<
     let mut penalties = vec![0; validator_count];
     if is_in_inactivity_leak(state, context) {
         let matching_target_attestations =
-            get_matching_target_attestations(state, previous_epoch, context)?
-                .cloned()
-                .collect::<Vec<_>>();
-        let matching_target_attesting_indices = get_unslashed_attesting_indices(
-            state,
-            matching_target_attestations.as_slice(),
-            context,
-        )?;
+            get_matching_target_attestations(state, previous_epoch, context)?.collect::<Vec<_>>();
```

Comment:
i may be sending you down the wrong rabbit hole but we have this patter now of "call get_matching_*_atts" and then `collect` them into a `Vec`

just so we can pipe something into `get_unslashed_attesting_indices` -- i have wonder if we can skip the `collect` by relaxing `get_unslashed_attesting_indices` so that it accepts anything that `impl Iterator<Item=&PendingAtt>`?

Reply:
Yeah, I think that logic makes sense.

So, if `get_unslashed_attesting_indices` is relaxed, then it probably makes sense to also relax `get_attestation_component_deltas` since `attestations` are simply passed from the `get_attestation_component_deltas` `attestations` param to `get_unslashed_attesting_indices`.